packages/catalog-search/src/SearchSuggestions.jsx

```diff
@@ -14,15 +14,18 @@ const SearchSuggestions = ({
   const getLinkToCourse = (course) => `/${enterpriseSlug}/course/${course.key}`;
   const getLinkToProgram = (program) => `/${enterpriseSlug}/program/${program.aggregation_key.split(':').pop()}`;
 
+  const courses = autoCompleteHits.filter(hit => hit.learning_type === 'course');
+  const programs = autoCompleteHits.filter(hit => hit.learning_type === 'program');
+  const execEdCourses = autoCompleteHits.filter(hit => hit.learning_type === 'Executive Education');
```

Comment:
It may be worth trying to refactor this to only do the iteration over `autoCompleteHits` once to compute the `courses`, `programs`, and `execEdCourses`, e.g.:

```jsx
const courses = [];
const programs = [];
const execEdCourses = [];
autoCompleteHits.forEach((hit) => {
  const { learning_type: learningType } = hit;

  if (hit.learner_type === 'course') { courses.push(hit); }
  if (hit.learner_type === 'program') { programs.push(hit); }
  if (hit.learner_type === 'Executive Education') { execEdCourses.push(hit); }
}));
```